spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -308,6 +339,17 @@ public Scan buildChangelogScan() {
     return new SparkChangelogScan(spark, table, scan, readConf, expectedSchema, filterExpressions);
   }
 
+  private Long getStartSnapshotId(Long startTimestamp) {
+    Snapshot oldestSnapshotAfter = SnapshotUtil.oldestAncestorAfter(table, startTimestamp);
```

Comment:
Spark processes `startTimestamp` logically differently from Flink. I wonder if we should unify them?

In Flink, it will first find whether there is a snapshot, its time is equal `startTimestamp`. If not, `SnapshotUtil.oldestAncestorAfter` will be used.

Reply:
Is it inclusive for the `startTimestamp` in Flink?
>Spark processes startTimestamp logically differently from Flink. I wonder if we should unify them?

Is it necessary to do that since `oldestAncestorAfter(startTimestamp)` will return the snapshot whose timestamp equals to `startTimestamp`. 
>In Flink, it will first find whether there is a snapshot, its time is equal startTimestamp. If not, SnapshotUtil.oldestAncestorAfter will be used.

Reply:
> Is it necessary to do that since oldestAncestorAfter(startTimestamp) will return the snapshot whose timestamp equals to startTimestamp.

I have re-checked and you are right. Then,  I raised https://github.com/apache/iceberg/pull/6401 to try to unify them.

Reply:
Yeah, I was confused by the name at the beginning as well.